src/pages/workspaces/workspace/Data.js

```diff
@@ -944,10 +948,28 @@ const WorkspaceData = _.flow(
   loadMetadata = withErrorReporting('Error loading workspace entity data', async () => {
     const { namespace, name, signal } = this.props
     const { selectedDataType } = this.state
-    const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata()
+
+    const [entityMetadata, { resources: snapshotMetadata }] = await Promise.all([
+      Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata(),
+      Ajax(signal).Workspaces.workspace(namespace, name).listSnapshot(1000, 0)
+    ])
+
+    const snapshotEntities = await Promise.all(_.map(({ name: snapshotName }) => {
+      return Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(namespace, snapshotName)
+    }, snapshotMetadata))
+
+    const snapshotDetails = _.reduce(
```

Comment:
Instead of `_.reduce` and dealing with indices, we could use `_.zip` to associate the two arrays, and then `_.map` and `_.fromPairs` to create the object. I think it would increase readability a little by using more specific (less powerful) tools.